src/app/(proper_react)/(redesign)/(authenticated)/admin/announcements/AnnouncementsAdmin.tsx

```diff
@@ -414,63 +411,81 @@ export const AnnouncementsAdmin = (props: Props) => {
       <AnnouncementsModal
         isOpen={isModalOpen}
         onClose={() => setIsModalOpen(false)}
-        notificationToEdit={activeAnnouncementToEdit}
+        announcementToEdit={activeAnnouncementToEdit}
         onAddAnnouncement={handleAddAnnouncement}
         onUpdateAnnouncement={handleUpdateAnnouncement}
+        isSubmitting={isSubmitting}
+        setIsSubmitting={setIsSubmitting}
       />
     </div>
   );
 };
 
 type LocalizedAnnouncementStringProps = {
-  notification: AnnouncementRow;
+  announcement: AnnouncementRow;
   type: "title" | "description" | "cta-label";
+  truncatedDescription?: boolean;
 };
 
 export const LocalizedAnnouncementString = (
   props: LocalizedAnnouncementStringProps,
 ) => {
+  function truncateDescription(str: string): string {
+    const MAX_CHARACTERS = 80;
+    if (str.length <= MAX_CHARACTERS) return str;
+    return str.slice(0, MAX_CHARACTERS) + "…";
+  }
```

Comment:
Truncating strings has come up in the past, and I remember that @flodolo had input on a couple of things to consider, but I could not find the reference. Because this is also used for user-facing views: Is it safe to truncate a string like this if it is `en`-only?

Reply:
Thanks for the input; that makes sense. @codemist Would it work with the design to have CSS handle the truncation with [text-overflow: ellipsis;](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow) or at least not truncate in the middle of a word?